raster-foundry/raster-foundry
# Streamline upload processing

## Overview

This PR:

- switches from a series of hand rolled cogification commands to `gdal_translate -of COG`
- removes some stale dependencies like the rf python client (lol)
- upgrades some deps to make rasterio and gdal agree about proj
- removes upload processing for anything but s3 uploads
- removes ingest for Landsat 8 and Sentinel-2 data

### Checklist


### Notes

~[wip] because I think I've tested what I need to? But there's so much removed that it's tough to feel confident about that so
I'll try again tomorrow morning.~ I made sure everything still imports, so at least there aren't any obvious missing symbols.

As a test of tile serving I ran the same `gdal_translate` command on one of the tifs in the data dir, uploaded it to s3, and pasted it into the ER staging app, and tiles were 🌶️ 

We should consider, if this is going to be the beginning of lighting things on :fire: (along with azavea/raster-foundry-platform#1093), creating a 2.0.0 milestone and collecting big deletion issues in it. I certainly have a few other things on the chopping block in my brain.

## Testing Instructions

- rebuild the batch container -- `docker-compose build batch`. This might take a while because the container received a significant upgrade.
- `./sbt ;api/assembly` if you haven't done so in a while
- bring up the API server
- here's a nice upload for you -- save it in `upload.json` with a tif that the batch container can see:

```json
{
	"uploadStatus": "UPLOADED",
	"fileType": "GEOTIFF",
	"uploadType": "S3",
	"files": ["<an s3 or local-to-batch-container-path path to a tif>"],
	"datasource": "e4d1b0a0-99ee-493d-8548-53df8e20d2aa",
	"metadata": {},
	"owner": null,
	"visibility": "PRIVATE",
	"projectId": null,
	"layerId": null,
	"source": null,
	"keepInSourceBucket": false,
	"annotationProjectId": null,
	"generateTasks": false
}
```

- get a JWT with the "RF dev user refresh token" in Lastpass -- `echo '{"refresh_token": "<the dev refresh token>"}' | http :9100/api/tokens`
- set the `JWT_AUTH_TOKEN` variable to the bearer token in the response
- create your upload: `cat upload.json | http --auth-type=jwt :9100/api/uploads | jq .id`
- process your upload -- `./scripts/console batch bash`, then `rf process-upload <the id>` -- it should work
- `rf <TAB><TAB>` to make sure autocomplete was removed for ingest


Closes azavea/raster-foundry-platform#1072
